etl/steps/data/meadow/lis/2023-01-18/luxembourg_income_study.py

```diff
@@ -0,0 +1,95 @@
+"""
+
+import numpy as np
+import pandas as pd
+from owid.catalog import Dataset, Table
+from structlog import get_logger
+
+from etl.helpers import PathFinder
+from etl.paths import REFERENCE_DATASET
+from etl.snapshot import Snapshot
+from etl.steps.data.converters import convert_snapshot_metadata
+
+# Initialize logger.
+log = get_logger()
+
+# Get paths and naming conventions for current step.
+paths = PathFinder(__file__)
+
+# Create a dictionary with the names of the snapshots and their id variables
+snapshots_dict = {
+    "lis_keyvars": ["country", "year", "dataset", "variable", "eq"],
+    "lis_abs_poverty": ["country", "year", "dataset", "variable", "eq", "povline"],
+    "lis_distribution": ["country", "year", "dataset", "variable", "eq", "percentile"],
+}
+
+
+def run(dest_dir: str) -> None:
+    log.info("luxembourg_income_study.start")
+
+    #
+    # Load inputs.
+    # Load reference file with country names in OWID standard
+    ds_reference = Dataset(REFERENCE_DATASET)
+    df_countries_regions = pd.DataFrame(ds_reference["countries_regions"]).reset_index()
+
+    for ds_name, ds_ids in snapshots_dict.items():
+
+        print(ds_name)
+
+        # Retrieve snapshot.
+        snap: Snapshot = paths.load_dependency(f"{ds_name}.csv")
+
+        # Load data from snapshot.
+        df = pd.read_csv(snap.path)
+
+        df[[col for col in df.columns if col not in ds_ids]] = df[
+            [col for col in df.columns if col not in ds_ids]
+        ].apply(pd.to_numeric, errors="coerce")
+
+        # Extract country and year from dataset
+        df["country"] = df["dataset"].str[:2].str.upper()
+        df["year"] = df["dataset"].str[2:4].astype(int)
+
+        # Replace "UK" with "GB" (official ISO-2 name for the United Kingdom)
+        df["country"] = np.where(df["country"] == "UK", "GB", df["country"])
+
+        # Create year variable in the format YYYY instead of YY
+        df["year"] = np.where(df["year"] > 50, df["year"] + 1900, df["year"] + 2000)
+
+        # Merge dataset and country dictionary to get the name of the country (and rename it as "country")
+        df = pd.merge(
+            df, df_countries_regions[["name", "iso_alpha2"]], left_on="country", right_on="iso_alpha2", how="left"
+        )
+        df = df.drop(columns=["country", "iso_alpha2"])
+        df = df.rename(columns={"name": "country"})
+
+        # Move country and year to the beginning
+        cols_to_move = ["country", "year"]
+        df = df[cols_to_move + [col for col in df.columns if col not in cols_to_move]]
+
+        #
+        # Process data.
+        #
+        # Create a new table and ensure all columns are snake-case.
+        tb = Table(df, short_name=ds_name, underscore=True)
+
+        #
+        # Save outputs.
+        #
+        # Create a new meadow dataset with the same metadata as the snapshot.
+        ds_meadow = Dataset.create_empty(dest_dir, metadata=convert_snapshot_metadata(snap.metadata))
```

Comment:
Please also take a look if the rest of the things make sense. I am just trying to upload the three snapshots into three tables in one dataset.

Reply:
Hey @paarriagadap, I pushed a [commit with a fix](https://github.com/owid/etl/pull/784/commits/717f8601394c01b107605b642f513053333153fd). You were almost there, but you had to create dataset only once and then call `ds_meadow.add(tb)` for all three tables. Also `paths.load_dependency("step_name")` always points to short name of a step, i.e.`paths.load_dependency("luxembourg_income_study")`.

Reply:
Thank you very much, @Marigold! So I guess it's OK to call the snapshot `lis_keyvars` twice (before the loop and inside it) to make the `Dataset.create_empty` work

Reply:
Up to you, I thought it would be better to call it again to make it transparent that we use metadata from `lis_keyvars` snapshot. (Calling it twice doesn't impact performance)